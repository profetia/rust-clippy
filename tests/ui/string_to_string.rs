#![warn(clippy::string_to_string)]
#![allow(clippy::redundant_clone)]

fn main() {
    let mut message = String::from("Hello");
    let mut v = message.to_string();
    //~^ string_to_string
}

mod issue14175 {
    fn string_behind_ref() {
        let mut v1 = String::from("Hello");
        let mut v2 = Some(&v1).map(|s| s.to_string());
        //~^ string_to_string
    }
}
