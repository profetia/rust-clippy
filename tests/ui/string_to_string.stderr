error: `to_string()` called on a `String`
  --> tests/ui/string_to_string.rs:6:17
   |
LL |     let mut v = message.to_string();
   |                 ^^^^^^^^^^^^^^^^^^^
   |
   = help: consider using `.clone()`
   = note: `-D clippy::string-to-string` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::string_to_string)]`

error: `to_string()` called on a `String`
  --> tests/ui/string_to_string.rs:13:40
   |
LL |         let mut v2 = Some(&v1).map(|s| s.to_string());
   |                                        ^^^^^^^^^^^^^
   |
   = help: consider using `.clone()`

error: aborting due to 2 previous errors

